engines/mtropolis/detection.h

```diff
@@ -104,6 +104,7 @@ struct MTropolisGameDescription {
 	int gameID;
 	int gameType;
 	MTropolisGameBootID bootID;
+	const char *mainFileWindows;
```

Comment:
This will need to be adapted once PR #5829 is merged.

Reply:
Is that concern still valid? I see no direct merge issues with the final version of PR #5829 that was merged.

Reply:
Yes - you'll need to replace `AD_GAME_DESCRIPTION_HELPERS(desc);` with helpers that make a copy of the string. This code (adapted from Mohawk) should do the job:

```
	uint32 sizeBuffer() const {
		uint32 ret = desc.sizeBuffer();
		ret += ADDynamicDescription::strSizeBuffer(mainFileWindows);
		return ret;
	}

	void *toBuffer(void *buffer) {
		buffer = desc.toBuffer(buffer);
		buffer = ADDynamicDescription::strToBuffer(buffer, mainFileWindows);
		return buffer;
	}
```

Reply:
For a more detailed information, you can look at the [Wiki page dedicated to this change](https://wiki.scummvm.org/index.php?title=HOWTO-Convert-Detection-Split#Changes_when_the_game_descriptor_is_of_custom_type_with_pointers_to_strings).

Reply:
Thanks for the clarification. I pushed the suggested changes.